src/settings/controllers/SlidingSyncController.ts

```diff
@@ -38,6 +93,9 @@ export default class SlidingSyncController extends SettingController {
         if (SettingsStore.getValue("feature_sliding_sync")) {
             return _t("labs|sliding_sync_disabled_notice");
         }
+        if (!this.slidingSyncHealthCheck()) {
```

Comment:
`slidingSyncHealthCheck` returns a promise so this will never be false

Reply:
Ah. I'll have to have a think about that one, given it's not in an async function.